specification/dartLangSpec.tex

```diff
@@ -11677,113 +12041,209 @@ \subsection{Identifier Reference}
-\item If $d$ is the declaration of a library variable, top-level getter or top-level setter, then evaluation of $e$ is equivalent to evaluation of the top level getter invocation (\ref{topLevelGetterInvocation}) \id.
-\item Otherwise, if $e$ occurs inside a top level or static function (be it function, method, getter, or setter) or variable initializer, evaluation of $e$ causes a \code{NoSuchMethod} to be thrown.
-\item Otherwise, evaluation of $e$ is equivalent to evaluation of the property extraction (\ref{propertyExtraction}) \code{\THIS.\id}.
+\item
+  If $D$ is a class, mixin, type alias or type parameter,
+  the static type of $e$ is \code{Type}.
+\item
+  If $D$ is the declaration of
+  a library variable or getter,
+  the static type of $e$ is the static type of the
+  top level getter invocation \id{}
+  (\ref{topLevelGetterInvocation}).
+\item
+  If $D$ is a static method, library function, or local function,
+  the static type of $e$ is the function type of $D$.
+\item
+  If $D$ is the declaration of a static variable or getter
+  and $D$ occurs in the class $C$,
+  the static type of $e$ is the static type of the getter invocation
+  (\ref{propertyExtraction})
+  \code{$C$.\id}.
+\item
+  If $D$ is a local variable declaration
+  (\commentary{which can be a formal parameter})
+  the static type of $e$ is the type of the variable $v$ declared by $D$,
+  unless $v$ is known to have some type $T$,
+  where $T$ is a subtype of any other type $S$
+  such that $v$ is known to have type $S$,
+  in which case the static type of $e$ is $T$.
+\item
+  \commentary{%
+    A lexical lookup will never yield a declaration $D$
+    which is an instance member.%
+  }
 \end{itemize}
+\vspace{-1ex}
+\EndCase
+
+\LMHash{}%
+\Case{Lexical lookup yields an import prefix}
+In this case the lexical lookup
+(\ref{lexicalLookup})
+for \id{} yields an import prefix $p$.
+% A prefix can never be used as a stand-alone expression.
+In this case a compile-time error occurs,
+unless the token immediately following $e$ is \lit{.}.
+No static type is associated with $e$ in this case.
+
+% A type is specified in \ref{imports}, so it would be consistent to report
+% it. However, no usage will be made of that type, because the static type
+% of every construct of the form $p$.something is specified without referring
+% to the static type of $p$. So we do not mention that type here.
+\commentary{%
+No such type is needed, because every construct where an import prefix $p$ is
+used and followed by \lit{.} is specified in such a way that the type
+of $p$ is not used.%
+}
+\EndCase
 
 \LMHash{}%
-The static type of $e$ is determined as follows:
+\Case{Lexical lookup yields a member signature}
+In this case the lexical lookup
+(\ref{lexicalLookup})
+for \id{} yields a member signature $s$.
+%
+In this situation $e$ is treated as
+(\ref{notation})
+\code{\THIS.\id}.
+
+\commentary{%
+This occurs when no declaration named \id{} is in scope,
```

Comment:
> This also occurs ...

Reply:
Resolved in the same way as the other location with similar text.